packages/calcite-components/src/components/carousel/carousel.tsx

```diff
@@ -0,0 +1,353 @@
+    }
+
+    const activeItemIndex = items?.findIndex((item) => item.active);
+    const requestedActiveIndex = activeItemIndex > -1 ? activeItemIndex : 0;
+
+    this.items = items;
+    this.setSelectedItem(false, requestedActiveIndex);
+  };
+
+  private setSelectedItem = (emit: boolean, requestedIndex: number): void => {
+    this.items?.forEach((el, index) => {
+      const isMatch = requestedIndex === index;
+      el.active = isMatch;
+      if (isMatch) {
+        this.selectedItem = el;
+        this.activeIndex = index;
+      }
+    });
+
+    if (emit) {
+      this.calciteCarouselChange.emit();
+    }
+  };
+
+  private nextItem = (): void => {
+    this.direction = "advancing";
+    const nextIndex = this.activeIndex === this.items?.length - 1 ? 0 : this.activeIndex + 1;
+    this.setSelectedItem(true, nextIndex);
+  };
+
+  private previousItem = (): void => {
+    this.direction = "retreating";
+    const previousIndex = this.activeIndex === 0 ? this.items?.length - 1 : this.activeIndex - 1;
+    this.setSelectedItem(true, previousIndex);
+  };
+
+  private keyDownHandler = (event: KeyboardEvent): void => {
+    if (event.target !== this.container) {
+      return;
+    }
+
+    switch (event.key) {
+      case "ArrowRight":
+        event.preventDefault();
+        this.nextItem();
+        break;
+      case "ArrowLeft":
+        event.preventDefault();
+        this.previousItem();
+        break;
+      case "Home":
+        event.preventDefault();
+        this.setSelectedItem(true, 0);
+        break;
+      case "End":
+        event.preventDefault();
+        this.setSelectedItem(true, this.items?.length - 1);
+        break;
+    }
+  };
+
+  private storeContainerRef = (el: HTMLDivElement): void => {
+    this.container = el;
+  };
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+  renderPagination(): VNode {
+    const { activeIndex } = this;
+    const itemGuid = guid();
+
+    return (
+      <div
+        class={{
+          [CSS.pagination]: true,
+          [CSS.isOverlay]: this.controlOverlay,
+        }}
+      >
+        {this.arrowType === "inline" && this.renderPreviousArrow()}
+        {this.items?.map((item, index) => (
+          <Fragment>
+            <calcite-action
+              appearance={this.controlOverlay ? "solid" : "transparent"}
+              class={`pagination-item${index === activeIndex ? " active-icon" : ""}`}
+              icon={index === activeIndex ? ICONS.active : ICONS.inactive}
+              id={`${itemGuid}-${index}`}
+              label={item.label}
+              onClick={() => this.setSelectedItem(true, index)}
+              scale="s"
+              text={item.label}
+            />
+            {!this.tooltipsDisabled && (
+              <calcite-tooltip placement="bottom" reference-element={`${itemGuid}-${index}`}>
```

Comment:
nitpick: use the dom element as the reference element if possible. Less lookup in the DOM this way and potential shadow dom issues.

Reply:
This would be the first component that uses a tooltip internally. Its been our pattern before to use a native `title` attribute for tooltips and let a user supply tooltips with slots if they need to. Should we discuss this?

Reply:
😮😮 that is good to know. that sounds like the more consistent direction to go?

Reply:
Nice catch. Agree with changing this to `title` attribute. Might also make sense to remove `tooltipsDisabled` too, like @macandcheese suggested, in this case.